packages/dds/tree/src/shared-tree-core/sharedTreeCore.ts

```diff
@@ -258,4 +261,23 @@ export type SummaryElementStringifier = (contents: unknown) => string;
 /**
  * Parses a serialized/summarized string into an object, rehydrating any Fluid handles as necessary
  */
- export type SummaryElementParser = (contents: string) => unknown;
+export type SummaryElementParser = (contents: string) => unknown;
+
+/** Compose an {@link IChannelStorageService} which prefixes all paths before forwarding them to the original service */
```

Comment:
I learned recently that [our styleguide](https://github.com/microsoft/FluidFramework/wiki/TSDoc-Guidelines) explicitly prohibits single-line TSDoc comments.
```suggestion
/**
 * Compose an {@link IChannelStorageService} which prefixes all paths before forwarding them to the original service 
 */
```

Reply:
*Grooaannn*. I'm okay with this for public/exported functions, but for private/non-external functions I like to just put a little explanatory comment, but I use the doc syntax rather than just a `//` comment because then VS code shows it when you mouse-over call site. @Josmithr thoughts on clarifying that this guideline could be for public but not private functions?

Reply:
The linter tooling can't really differentiate (not being enforced yet, but will be soon), but I think the main point is to promote consistent style for general legibility of the code, rather than something special for public API surfaces (my 2 cents).

Reply:
Yeah, I have the same preferences.

Reply:
@Josmithr but will the linter complain about a single-line `//` comment rather than a `/**` comment? Because that's what I'm going to do instead, and it seems silly to lose a nice VS code feature (mouse over for description) for no reason.

Reply:
I mean, the reason is consistency and legibility. It's the same reason we have other linter settings. So, I don't think "for no reason" is a fair assessment. Is there a reason not to just use the multi-line syntax in this case? Consistent styling can have a big impact on overall legibility / accessibility of our code.

No, the linter won't care about `//` comments, but I don't think 1 line vs multi-line should be the reason to use one or the other. If the comment info is only useful to someone looking directly at the code, then a `//` comment is probably appropriate, but if someone using the API should know about it, we should use TSDoc syntax.

Once the linter tooling is turned on, this should also be auto-fixable, so if your concern is having to always type out the multi-line syntax, that should be mitigated at least somewhat.

Again, just my 2 cents 😊

Reply:
Sure, it's subjective. I like to be thorough and put a comment on most helper functions that I write, but in a file with several helper functions, some which might be nested, etc., having an extra two lines of (essentially) whitespace for every single one actually does start to push things around and make the code taller, require more scrolling, etc. I guess my personal heuristic is balanced such that I often think "I should add a little comment here to flesh out what this method does, it's only one more line. If it were three lines though, that's more annoying than it is helpful". Maybe I'm just weirdly particular. Regardless, there is a data point for you in case it ever comes up again ("Noah does not like") :)

Reply:
That's fair. There is definitely a trade-off in terms of screen real estate. And noted (or "Noah-ted", if you will) 😁